oeps/architectural-decisions/oep-0066-arch-role-based-access-control.rst

```diff
@@ -0,0 +1,88 @@
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2023-08-16
+   * - Review Period
+     - TBD
+
+*******
+Context
+*******
+
+There are currently multiple ways to assign roles to users at multiple context levels (course, organization, instance). 
+These allow for assigning a role to a user, but do not indicate what the usages of the role are or
+allow for any modification of the roles themselves. 
+
+There are currently two distinct systems for adding course level roles that are comingled in the UI.
+The database tables involved in the first system (course access roles) is the student_courseaccessrole table 
+and the database tables involved in the second system (discussion roles) are the django_comment_client_role table and associated tables.
+Additionally, it is possible to grant a user access to a course using the Django Admin auth_permission table and associated tables.
+
+The current systems require developer work across multiple repos, MFEs, and IDAs in order 
+to add a new role, modify an existing role, or remove an existing role.
+
+Additional flexibility from the system has been requested in order to add new roles, 
+modify roles, create custom roles, understand role impact, and bulk add and remove role assignments.
+The current systems do not allow for customization and user controlled modification of existing roles.
+
+
+Decision
+********
+
+We will create a new system that utilizes permissions assigned to roles
+which are assigned to users to grant access to different portions of the system.
+This system will replace the two existing systems, course access roles and discussion roles. 
+The new system will initially be rolled out for 1 - 3 new roles, but once it is proven 
+effective, it will be rolled out for the roles that exist in course access roles and discussion roles. 
+At that time we will sunset the usage of course access roles and discussion roles.
+
+Consequences
+************
+
+Once completed this will remove one level of complexity from the overall authorization picture 
+by decreasing the systems involved by one, however during the interim it will add complexity by increasing 
+the number of involved systems and requiring all access checks to check for a role and a permission.
+
+After implementation, access checks will check for a permission instead of a role. This will mean that 
+if a new permission is added, each default role will need to be updated to indicate if it grants the new 
+permission. It will also mean that if a new role is added, there will be no need to update each of the services 
+that grant access as they will continue to check for the same permissions no matter how many roles grant those permissions.
+
+Adding this new system will allow for eventual customized roles on an organization basis. 
+This system will continue to allow for assigning a course level role to a user at 
+an organization or instance level. 
+
+References
+**********
+
+.. image:: oep-0066/rbac_overview.png
```

Comment:
This image contains much more detail that is not in this document. Is there another doc that contains this information? Do you want a sub-document or other sections? For accessibility purposes (which would probably help most readers), having this information in doc form would be useful. We could discuss options.

Also, this image contains so much. Even just breaking it into 2 images, rather than one that is so wide, would help give each more room.